java/code/src/com/redhat/rhn/frontend/xmlrpc/channel/software/ChannelSoftwareHandler.java

```diff
@@ -2331,6 +2331,32 @@ private void channelAdminPermCheck(User loggedInUser) {
      */
     public Object[] mergePackages(User loggedInUser, String mergeFromLabel,
             String mergeToLabel) {
+        return mergePackages(loggedInUser, mergeFromLabel, mergeToLabel, false);
```

Comment:
Is there any good reason to call this with `false` in future? Will `true` break the functionality under some conditions?

If `true` should work in general, we should just do it always and do not implement an extra function.

Reply:
This is a bit tricky topic, but I'd say it shouldn't be a default behavior. I think, that in situations when user wants to merge 1 package from a modular channel into another one, it does not make sense to align the modular metadata (unless user really wants it, in which case they can use the boolean parameter). In any case, using `mergePackages` on modular channels can be somewhat risky: let's say that both source and target channels have incompatible modular data. Blindly aligning the modular data could break the target channel...


> What about cloneByDate tool?

This tool uses `CloneChannelCommand` in the end -> the modular metadata is cloned as well. Next question would be: is the modular repodata valid if the target channel only contains subset packages of the source? I think using this tool for modular channels can bring some funky results.

Reply:
Ok, I just wanted to double check if this is really needed.

About CloneByDate: should we prevent using this tool for modular channels? Should we document that it would be better to not use it?